pypdf/_writer.py

```diff
@@ -852,6 +852,11 @@ def clone_reader_document_root(self, reader: PdfReader) -> None:
         self._root = self._root_object.indirect_reference  # type: ignore[assignment]
         self._pages = self._root_object.raw_get("/Pages")
         self._flatten()
+        for p in self.flattened_pages:
+            o = self._objects[p.idnum - 1].get_object()
+            self._objects[p.idnum - 1] = PageObject(self, p)
```

Comment:
Why did you add this line?

Reply:
do you mean L857 ? if it is to create Pages. The cloning process works at low level and duplicates an DictionaryObject into a DictionaryObject.